Update commissioning macros

BrieucF opened the pull request:
Here are the main changes I made to the QCD commissioning macros :
- CommPlotProducer::Loop takes now what we often want to change as arguments (trigger, jet pt cut, PU reweighting file, ...)
- It is easier to run in parallel thanks to the runCode_parallel.C
- launchRunCodeParallel.py is simply an example on how it can be used. It is specific to my local cluster, let me know if you want me to remove it from the PR.

mverzett commented:
Hi Brieuc,

thanks for the PR. Few comments/questions:
- what does runCode_parallel.C specifically? I don't have clear in mind what is your workflow, could you please detail it a bit more for my understanding?
- Regarding launchRunCodeParallel.py it is indeed very environment-specific. Could you please factor all the non-specific part at the beginning of the script and add a cluster specific switch at the bottom of the script? You can take some ideas from [here](https://github.com/mverzett/.bin/blob/master/host.py). If you go down this road remember to raise an exception with meaningful text in case the host is not recognised.

Thanks!

BrieucF committed:
Update commissioning macros

BrieucF commented:
- The general workflow is the following :
  - Obtain the btag n-tuple with the BTagAnalyzer
  - Get TH1's for data and mc (splitted in jet flavour) from these n-tuples using `CommPlotProducer.C` members. `runCode_parallel.C` is an example of interface between the user and `CommPlotProducer.C` facilities. It calls the functions in the proper order and propagate to `CommPlotProducer::Loop` the user defined arguments (see detailed explanation of the arguments below). It is not intended to be used out of the box, it has to be adapted per user/production.
  - Finally, draw the data/mc stacked plot with DrawCommPlot.C and runPlot.C (untouched by this PR).
- I removed the `launchRunCodeParallel.py` from the PR because it was simply one more layer to ease job submission in Louvain cluster (condor based), not easily generalisable. The 'minimal working example' is actually already `runCode_parallel.C` which can be called directly with something like
  `root -b -q runCode_parallel.C\(TString charForChain, Int_t nFile, TString outFileName, bool isMuEn, Int_t TrigerPt, Int_t jetPtMin, Int_t jetPtMax, TString PUreweightingFile = "", bool officialPU = false)` 
  replacing arguments by appropriate value. 
  Here is the detailed explanation of the arguments : 
  - `charForChain` and `nFile` specify the input btagana n-tuple. They talk to each other : either you give _one_ input file and nFile==0, or you give the input file pattern (string such as "my/local/dir/JetTree_mc_INDEX.root") and the number of file you want to run on. "INDEX" will be replaced in a loop by the root file number and added to the TChain (this is usefull if you want one job to run over multiple files).
  - `outFileName` is the name of the root file (without '.root') output which will contain the TH1's.
  - `isMuEn` is a boolean flag which allows to use the same code to run on MuonEnriched or Inclusive QCD.
  - `TrigerPt`, `jetPtMin`, `jetPtMax` are quite explicit
  - `PUreweightingFile` is the root file path one wants to use to apply PU reweighting based on nPV. To have a "proper" PU reweighting, one has to run twice on the btagana n-tuples : first time setting `PUreweightingFile` to "" which will produce the TH1's without any PU reweighting. Then, after having hadded all the rootFiles to have all mc pthat bins and data in one rootfile, one re-run by setting `PUreweightingFile` to the latter.
  - `officialPU` is set to true if we want to apply the official PU reweighting. It is technically implemented but never worked so far due to the problem deriving PU profile for pre-scaled triggers.

Last thing to know is that one has to set the number of entries we have before selection for each pt hat bin using m.Fill_nevent(...) in `runCode_parallel.C`. To cross check that it was properly filled, one can look at the pthat distribution after having hadded the different pt hat bins.

ghost commented:
Dear Brieuc,

many thanks for the PR. This is useful, @kderoove and @carolinecollard  can then work on top of this to apply additional updates. Would you mind updating:

nHitPixCut

to 1? Which is the new default (not zero)? Thanks.

Could you also cross check if the 3,919 lines difference in CommPlotProducer.C is expected? Is some difference in the indentation? Could you check what happened.

Many thanks.

Cheers,

Ivan

BrieucF commented:
PR updated. The "full file" diff was due to different line break, this is solved now.

mverzett commented:
Dear Brieuc,
Thanks for the changes and the explaination. One last request: could you please port this request to the 8.1.X branch, before we loose sync?

Thanks
